prime-router/src/main/kotlin/azure/LookupTableFunctions.kt

```diff
@@ -149,64 +188,71 @@ class LookupTableFunctions(
         ) request: HttpRequestMessage<String?>,
         @BindingName("tableName") tableName: String
     ): HttpResponseMessage {
-        return getOktaAuthenticator(PrincipalLevel.SYSTEM_ADMIN).checkAccess(request) { oktaAuthenticatedClaim ->
-            val inputData: List<Map<String, String>>
-            try {
-                val forceTableToLoad = request.queryParameters[forceQueryParameter].toBoolean()
-                inputData = mapper.readValue(request.body!!.toString())
-                if (inputData.isEmpty())
+        // Do authentication
+        val claims = AuthenticatedClaims.authenticate(request)
+            ?: return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
+        // Do authorization
+        if (!authorizedForLookupWrite(claims)) {
+            return HttpUtilities.unauthorizedResponse(request, authorizationFailure)
+        }
+        logger.info("User ${claims.userName} is authorized for endpoint ${request.uri}")
+
+        val inputData: List<Map<String, String>>
```

Comment:
Not required. Could move this `val` decl into the place it's assigned below in the try block.

Reply:
done.